solidity/contracts/bridge/Bridge.sol

```diff
@@ -1257,6 +1352,82 @@ contract Bridge is Ownable {
         bank.transferBalance(treasury, outputsInfo.totalTreasuryFee);
     }
 
+    /// @notice Submits a fraud challenge indicating that a UTXO being under
+    ///         wallet control was unlocked by the wallet but was not used
+    ///         according to the protocol rules. That means the wallet signed
+    ///         a transaction input pointing to that UTXO and there is a unique
+    ///         sighash and signature pair associated with that input. This
+    ///         function uses those parameters to create a fraud accusation that
+    ///         proves a given transaction input unlocking the given UTXO was
+    ///         actually signed by the wallet. This function cannot determine
+    ///         whether the transaction was actually broadcast and the input was
+    ///         consumed in a fraudulent way so it just opens a challenge period
+    ///         during which the wallet can defeat the challenge by submitting
+    ///         proof of a transaction that consumes the given input according
+    ///         to protocol rules. To prevent spurious allegations, the caller
+    ///         must deposit ETH that is returned back upon justified fraud
+    ///         challenge or confiscated otherwise.
+    /// @param walletPublicKey The public key of the wallet in the uncompressed
+    ///        and unprefixed format (64 bytes).
+    /// @param sighash The hash that was used to produce the ECDSA signature
+    ///        that is the subject of the fraud claim. This hash is constructed
+    ///        by applying double SHA-256 over a serialized subset of the
+    ///        transaction. The exact subset used as hash preimage depends on
+    ///        the transaction input the signature is produced for. See BIP-143
+    ///        for reference.
+    /// @param v Signature recovery value.
+    /// @param r Signature r value.
+    /// @param s Signature s value.
+    function submitFraudChallenge(
+        bytes memory walletPublicKey,
+        bytes32 sighash,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external payable {
+        bytes memory compressedWalletPublicKey = walletPublicKey
+            .compressPublicKey();
+        bytes20 walletPubKeyHash = bytes20(compressedWalletPublicKey.hash160());
+        require(
+            wallets[walletPubKeyHash].state == WalletState.Active ||
+                wallets[walletPubKeyHash].state == WalletState.MovingFunds,
+            "Wallet is neither active nor is moving funds"
+        );
+        fraudData.submitFraudChallenge(walletPublicKey, sighash, v, r, s);
```

Comment:
Let's add some empty lines between sections to improve readability.

Reply:
Done in bb8535061e0a713f9c5e478694d50df1d3fe3138.